contracts/KetherNFTRender.sol

```diff
@@ -21,26 +21,65 @@ contract KetherNFTRender is ITokenRenderer {
       '</g></svg>')));
   }
 
+  // Thanks to @townsendsam for giving us this reference https://gist.github.com/townsendsam/df2c420accb5ae786e856c97d13a2de6
+
+  function _generateAttributes(uint x, uint y, uint width, uint height, bool NSFW, bool forceNSFW) internal pure returns (string memory) {
+    return string(abi.encodePacked(
+      '[',
+         '{',
+            '"trait_type": "X",',
+            '"value": ', x.toString(),
+          '},',
+          '{',
+              '"trait_type": "Y",',
+              '"value": ', y.toString(),
+          '},',
+          '{',
+              '"trait_type": "Width",',
+              '"value": ', width.toString(),
+          '},',
+          '{',
+              '"trait_type": "Height",',
+              '"value": ', height.toString(),
+          '},',
+          '{',
+              '"trait_type": "Pixels",',
+              '"value": ', (height * width).toString(),
+          '},',
+          '{',
+              '"trait_type": "NSFW",',
+              '"value": ', _boolToString(NSFW),
```

Comment:
@mveytsman should this be `NSFW || forceNSFW`? (ie. effective NSFW value rather than internal NSFW value)

Reply:
As I understand it, having both is to let people differentiate images that are declared NSFW by the owner, versus ones that are marked NSFW by the admin.

I think in this case we want to show `NSFW` and `forceNSFW || NSFW` as the two traits, instead of `forceNSFW` and  `forceNSFW || NSFW`

Not sure what the names should be

`NSFW` and `NSWF (effective)`?

Reply:
I don't wanna have 3 attributes just for NSFW (NSFW, forcedNSFW, effective NSFW).

I think it's important to know the forcedNSFW state because that could effect the value of the plot.

It's important to know the effective NSFW for rendering purposes.

I don't think it's important to know the internal NSFW because that's subjective to change.

Reply:
👍